src/constructs/dns/dns-records.test.ts

```diff
@@ -0,0 +1,40 @@
+import "@aws-cdk/assert/jest";
+import { SynthUtils } from "@aws-cdk/assert";
+import { simpleGuStackForTesting } from "../../utils/test";
+import { GuCname, GuDnsRecordSet, RecordType } from "./dns-records";
+
+describe("The GuDnsRecordSet construct", () => {
+  it("should create the correct resources with minimal config", () => {
+    const stack = simpleGuStackForTesting();
+    new GuDnsRecordSet(stack, "TestRecord", {
+      name: "banana.example.com",
+      recordType: RecordType.CNAME,
+      resourceRecords: ["apple.example.com"],
+    });
+    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
+  });
+
+  it("should allow users to override the default TTL", () => {
+    const stack = simpleGuStackForTesting();
+    new GuDnsRecordSet(stack, "TestRecord", {
+      name: "banana.example.com",
+      recordType: RecordType.CNAME,
+      resourceRecords: ["apple.example.com"],
+      ttl: 123,
+    });
+    expect(stack).toHaveResourceLike("Guardian::DNS::RecordSet", {
+      TTL: 123,
+    });
+  });
+});
+
+describe("The GuCname construct", () => {
+  it("should create the correct resources with minimal config", () => {
+    const stack = simpleGuStackForTesting();
+    new GuCname(stack, "TestRecord", {
```

Comment:
Perhaps it's overkill to have a separate `GuCname` class, but it allows us to eliminate a bit of boilerplate, as shown here